frontend/src/components/Layout/AppBar/SearchField.vue

```diff
@@ -22,13 +23,23 @@ const searchQuery = computed({
     return route.query.q?.toString() || '';
   },
   set(value: string) {
-    if (value === '' || !value) {
-      router.back();
-    } else if (searchQuery.value) {
-      router.replace({ path: '/search', query: { q: value } });
-    } else {
-      router.push({ path: '/search', query: { q: value } });
-    }
+    router.replace({
+      ...router.currentRoute,
+      query: {
+        q: value || undefined
```

Comment:
```suggestion
        q: value.trim() || undefined
```

Just an easy QOL change